libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -382,45 +407,47 @@ data HOLE a b where
 
 toCtx ::
```

Comment:
Adding the string argument here is not how the `MonadGenError` is meant to be used. The argument `termstring` appears to always be `show` on the `Term` argument at the callsites. If this is the case, then why not just add a show constraint? If this is not the case, consider adding the extra logging information in a call to `explain` at the _call site_ of `toCtx`.

Either way, I would strongly prefer the use site to be wrapped in `explain` instead of this solution, as it pushes information down in the functions and makes them more complicated.